include/grid.h

```diff
@@ -273,6 +273,13 @@ class region_builder {
 		m_boxes.push_back(box);
 	}
 
+	void add(const box_vector<Dims>& boxes) & {
+		m_boxes.reserve(m_boxes.size() + boxes.size());
+		for(const auto& b : boxes) {
```

Comment:
Maybe have a comment here that we use a loop rather than `insert(end)` to skip over empty boxes, something which `add(region)` does not need to do.

[only if my comment below does not apply]